### PR TITLE
Made changes to fitting perspective for sesans data

### LINKED ISSUE #1932: Fitting details panel not displaying correct information when SESANS data loaded
The fitting details panel at the bottom of the FitPage normally displays the minimum and maximum values of the data range, and whether the data contains dI uncertainties from smearing.

When SESANS (.ses) data is loaded, there are some oddities:

- though the data is real space, the min/max values still show inverse units;
- the min value remains unchanged at the FitPage default (0.0005 Ang-1), though if the min value is altered in Fitting Options it does change;
- the max value is set to the largest spin-echo length in the data file;
- Smearing is announced as 'Use dQ Data', possibly because the file contains more than 3 columns so the third column is being interpreted as dI (it is actually dP).

Prior to loading data:
image

After loading sphere2micron.ses:
image

Suggestions:
The min spin-echo length in the file should be displayed when the file is loaded.
The .ses format does actually carry the units of spin-echo length and this could be used to ensure that whatever value is read is correctly converted to Angstroms.
If dP is read this should be stated.

### PR DESCRIPTION
**Pull request by caitwolf**
## Description

This PR reformats pieces of the fitting widget to accommodate SESANS data. This includes:
1. "Fitting Details" shows min and max spin echo lengths with units of Angstroms rather than inverse Angstroms.
2. "Fitting Details" shows "Hankel Transform" as the smearing function. Currently there are no resolution functions available for SESANS and the Hankel transform from I(q) to SESANS is shown as a resolution function. In the previous version this showed "Using dQ data" but performed the Hankel transform under the hood. This update makes it clear to the user that the Hankel transform is implemented and no other (unknown) resolution functions are applied.
3. The Fit Options tab similarly shows units of Angstroms rather than inverse Angstroms and correctly displays the min and max spin echo lengths as default. The weighting section also shows dP rather than dI (SESANS polarization rather than I(q)). 
4. The dropdown box in the Resolution tab is disabled. There are no other resolution functions available for SESANS (see point 1 above).

Fixes #1932 

## How Has This Been Tested?

Example SESANS, 1D and 2D have been loaded into the GUI then sent to the fitting panel to ensure display changes only occur for SESANS data.

## Review Checklist (please remove items if they don't apply):

- [ ] Code has been reviewed
- [ ] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [ ] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [ ] User documentation is available and complete (if required)
- [ ] Developers documentation is available and complete (if required)
- [ ] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

**Comment by caitwolf** [comment]
@lucas-wilkins I ended up making significant changes on this since your review; would you be able to take another look?

**Comment by caitwolf** [comment]
@butlerpd In the linked issue you mentioned 6.0.0. Should this be pointing to that branch instead?

File: src/sas/qtgui/Perspectives/Fitting/FittingWidget.py [hunk 1]
```diff
@@ -1577,6 +1597,27 @@ def onSelectionChanged(self):
                 return
             self.communicate.statusBarUpdateSignal.emit(update_text)
 
+    def onSesansData(self):
+        """
+        Updates the fitting widget format when SESANS data is loaded.
+        """
+        # update the units in the 'Fitting details' box of the Model tab on the Fit Panel
+        self.label_17.setText("<html><head/><body><p>Å</p></body></html>")
```
[the remark this answers is not in the document]
Comment (caitwolf) [reply src/sas/qtgui/Perspectives/Fitting/FittingWidget.py]: Good catch, they are not needed so I removed them

File: src/sas/qtgui/Perspectives/Fitting/FittingWidget.py [hunk 2]
```diff
@@ -528,6 +531,23 @@ def disableStructureCombo(self):
         self.lblStructure.setEnabled(False)
         self.enabled_sfmodel = False
 
+    def enableBackgroundParameter(self, set_value=None):
```
[the remark this answers is not in the document]
Comment (caitwolf) [reply src/sas/qtgui/Perspectives/Fitting/FittingWidget.py]: adding type hints for the enable and disable background parameter functions

File: src/sas/sascalc/fit/qsmearing.py [hunk 3]
```diff
@@ -56,14 +56,8 @@ def smear_selection(data, model = None):
...
-    #if data.dx is not None and data.meta_data['loader']=='SESANS':
-    if data.dx is not None and data.isSesans:
-        #if data.dx[0] > 0.0:
-        if np.size(data.dx[data.dx <= 0]) == 0:
-            _found_sesans = True
-        # if data.dx[0] <= 0.0:
-        if np.size(data.dx[data.dx <= 0]) > 0:
-            raise ValueError('one or more of your dx values are negative, please check the data file!')
+    if data.isSesans:  # data.dx data is not required in the Hankel transform for SESANS data
+        _found_sesans = True
```
[the remark this answers is not in the document]
Comment (caitwolf) [reply src/sas/sascalc/fit/qsmearing.py]: Confirmed this wasn't being used anywhere else in the code and removed

**Comment by caitwolf** [comment]
@butlerpd should we merge this change with the 6.0.0 release branch? This had gotten merged to main during contributor camp and I hadn't circled back to our above discussion to merge it in both

**Review by butlerpd — COMMENTED** [review]
Posted in error

**Pull request merged**